### ISSUE TITLE
Issue with missing DLL in runtime:4.8-windowsservercore-ltsc2022

### ISSUE THREAD

**Issue by nitinghaghada** [microsoft/Windows-Containers#465]
I am using mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022 as base image in my docker file to create custom image. After installing this custom image, I am frequently getting missing performance counter DLL error (Win32 error code 126) in the container in the event logs. See below error even log,

EventID : 1023
MachineName : XXXXXXXXX
Data : {}
Index : 4592
Category : (0)
CategoryNumber : 0
EntryType : Error
Message : Windows cannot load the extensible counter DLL "C:\Windows\System32\perfts.dll" (Win32 error code 126).
Source : Microsoft-Windows-Perflib
ReplacementStrings : {C:\Windows\System32\perfts.dll, 126}
InstanceId : 1023

When I looked for perfts.dll, I could not found it in c:\windows\system32 folder. Is it issue or removed this DLL in this version of Windows?

Thanks & regards,
Nitin

**Comment by fady-azmy-msft** [comment]
The perfts.dll is not available in the servercore image, may I ask what are you trying to do?

**Comment by nitinghaghada** [comment]
Thanks for reply. We are not doing anything with perfts.dll but because of missing DLL file, system frequently register errors. If it is not part of the core image then curious to know why system is trying to load it. Please note that my application running on container do not use performance counter or any processes which use this assembly.
I also found this file entry in system registry. 

Regards,
Nitin

**Comment by ntrappe-msft** [comment]
Hi, we'll need some more information from you to figure out what's going wrong on your end. I tried to hit the same error when creating a custom image from that base image. Unfortunately, I did not hit that error and everything worked for me. You can see the steps I took below.

### Provide the Following
- Host OS
- Paste the output of `docker inspect <image-id>` _(you can get the image ID by doing `docker images`)_
- List of repro steps

### Steps I Took
1. Using a Windows Server 2022 host with Docker Desktop.
2. Pulled down the image. 
```ruby
docker pull mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022
```
3. Spun up the container to test if it would work.
```ruby
docker run -it mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022
```
4. Created the following Dockerfile using that image to run my .NET app.
```dockerfile
# Use .NET base image
FROM mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022

# Set working directory
WORKDIR /app

# Copy application code over (includes .dll and .exe)
COPY /MyProgram /app

# Throwaway command for build process
RUN echo "done"
```
5. Build custom image using .NET base.
```
docker build -t my-dotnet .
```
6. Start up a container using custom image.
```
docker run -it my-dotnet
```
7. Run .NET app in container or do other basic actions.

**Comment by ntrappe-msft** [comment]
Hi, we'll need some more details to understand what's going wrong on your end. I tried to repro a scenario I thought would be similar to yours but I never found a similar perfts.dll error in my event logs.

### Info We Need
- Host OS
- Output of `docker info`
- Output of `docker inspect <image ID for mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022>` _(you can get the imageID by doing a `docker images`)_
- Repro steps

### Steps I Took
1. Using Windows Server 2022 host and Docker Desktop.
2. Pull the image.
```ruby
docker pull mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022
```
3. Start up a container with the image to check that it loads normally.
```ruby
docker run --rm -it  mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022
```
4. Create a simple Dockerfile using that as the base image.
```dockerfile
# Use .NET as base image
FROM mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2022

# Set working directory
WORKDIR /app

# Copy application code over (like .exe and .dll)
COPY /MyProgram /app

# Throwaway command for build
RUN echo "done"
```
5. Build our custom image.
```ruby
docker build -t my-dotnet .
```
6. Start up a container using the custom image.
```ruby
docker run --rm -it my-dotnet
```
7. While the container is still running, open a new PowerShell window and query the event logs for a similar error.
```ruby
Get-EventLog -LogName Application | Where-Object {$_.Message -like "*perfts.dll*"}
Get-EventLog -LogName Application | Where-Object {$_.Message -like "*cannot load* *DLL*"}
```

> [!NOTE]
> The first command checks the event logs for any entry that contains the word `perfts.dll`. The second checks for any entry with the phrases `cannot load` and `DLL`. Neither hit a match.

**Comment by nitinghaghada** [comment]
Hello Nicole,

Thank you for detail test and elaborate each steps. I have created another image from same windows server core base image (runtime:4.8-windowsservercore-ltsc2022) and didn't face issue. 

Issue is only with our custom docker image, We have enabled some IIS features and also adding .net core runtime and URL rewrite modules etc. It might causing that DLL issue.

Thanks for help. I will investigate it on our repo. We can close this issue. 

Thanks & regards,
Nitin
